service/worker/batcher/workflow.go

```diff
@@ -304,7 +333,19 @@ func startTaskProcessor(
 
 			switch batchParams.BatchType {
 			case BatchTypeTerminate:
-				err = processTerminateTask(ctx, limiter, task, batchParams, client)
+				err = processChildRecursiveTask(ctx, limiter, task, batchParams, client,
```

Comment:
So looking at this further.. Everything can go through the same method. When `applyOnChild` is null, it will only act on current, else will recursively look at children. So even signal can go through same path with `applyOnChild = null`